src/app/components/home-app/home-app.component.ts

```diff
@@ -86,99 +99,131 @@ export class HomeAppComponent implements OnDestroy, OnInit {
+            // observed
+            if (this.selectedSection === Section.archive) {
+              this.xApiService
+                .observedExhibitArchive(this.exhibitId, this.selectedTeamId)
+                .pipe(take(1))
+                .subscribe();
+            } else if (this.selectedSection === Section.wall) {
+              this.xApiService
+                .observedExhibitWall(this.exhibitId, this.selectedTeamId)
+                .pipe(take(1))
+                .subscribe();
+            }
+          } else {
+            // viewed
+            if (this.selectedSection === Section.archive) {
+              this.xApiService
+                .viewedExhibitArchive(this.exhibitId)
+                .pipe(take(1))
+                .subscribe();
+              if (cardId && cardId !== 'all') {
+                this.xApiService
+                  .viewedCard(this.exhibitId, cardId)
+                  .pipe(take(1))
+                  .subscribe();
+              }
+            } else if (this.selectedSection === Section.wall) {
+              this.xApiService
+                .viewedExhibitWall(this.exhibitId)
+                .pipe(take(1))
+                .subscribe();
             }
-          } else if (this.selectedSection === Section.wall) {
-            this.xApiService.viewedExhibitWall(this.exhibitId).pipe(take(1)).subscribe();
           }
         }
-      }
-    });
+      });
     // subscribe to the collections
-    this.collectionQuery.selectAll().pipe(takeUntil(this.unsubscribe$)).subscribe(collections => {
-      this.collectionList = collections;
-    });
+    this.collectionQuery
+      .selectAll()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((collections) => {
+        this.collectionList = collections;
+      });
     // subscribe to the exhibits
-    this.exhibitQuery.selectAll().pipe(takeUntil(this.unsubscribe$)).subscribe(exhibits => {
-      this.exhibitList = exhibits;
-    });
+    this.exhibitQuery
+      .selectAll()
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((exhibits) => {
+        this.exhibitList = exhibits;
+      });
     // subscribe to the active exhibit
-    (this.exhibitQuery.selectActive() as Observable<Exhibit>).pipe(takeUntil(this.unsubscribe$)).subscribe(exhibit => {
-      if (exhibit && exhibit.id) {
-        this.exhibit = exhibit;
-        this.collectionId = this.exhibit.collectionId;
-        this.currentMove = this.exhibit.currentMove;
-        this.currentInject = this.exhibit.currentInject;
-      };
-    });
+    (this.exhibitQuery.selectActive() as Observable<Exhibit>)
+      .pipe(takeUntil(this.unsubscribe$))
+      .subscribe((exhibit) => {
+        if (exhibit && exhibit.id) {
+          this.exhibit = exhibit;
+          this.collectionId = this.exhibit.collectionId;
+          this.currentMove = this.exhibit.currentMove;
+          this.currentInject = this.exhibit.currentInject;
+          this.handleExhibitMoveOrInjectChange();

```

Comment:
We need to compare the new values of id, currentMove, and currentInject to the values on this.exhibit and only call handleExhibitMoveOrInjectChange() when one of those values has changed.

There also needs to be a change in home-app.component.ts or signalr.service.ts to make sure that the switchTeam method gets called after the initial join() method call.

Reply:
I modified the logic for changeTeams so that it gets called on initialization. I am not sure I understand how the team is actually changed after trying to do it. I am also not sure what to do about exhibitId. It seems like the code is designed to set that only once when you load the HomeApp. If it changes then there are probably other things to do then just update the move and inject. Again, I might not have this right still.